pennylane/compiler/python_compiler/transforms/quantum/measurements_from_samples.py

```diff
@@ -564,24 +564,30 @@ def _get_static_shots_value_from_first_device_op(module: builtin.ModuleOp) -> in
     # The number of shots is passed as an SSA value operand to the DeviceInitOp
     shots_operand = device_op.shots
     shots_extract_op = shots_operand.owner
-    assert isinstance(shots_extract_op, tensor.ExtractOp), (
-        f"Expected owner of shots operand to be a tensor.ExtractOp but got "
-        f"{type(shots_extract_op).__name__}"
-    )
 
-    # This should be a stablehlo.ConstantOp that stores the values of `shots`
-    shots_constant_op = shots_extract_op.operands[0].owner
-    shots_value_attribute: builtin.DenseIntOrFPElementsAttr = shots_constant_op.properties.get(
-        "value"
-    )
-    assert (
-        shots_value_attribute is not None
-    ), "Cannot get number of shots; the constant op has no 'value' attribute"
+    # This branch is for a stablehlo.ConstantOp that stores the values of `shots`
```

Comment:
I think this is unnecessary and its also not true 😅 This branch is for `ExtractOp`. 

```suggestion
```